driver/src/main.rs

```diff
@@ -277,6 +277,11 @@ fn main() {
     let mut scheduler = options
         .scheduler
         .create(&*contract, &driver, scheduler_config);
+    // Retrieve the orderbook once because the event based orderbook needs to perform an expensive
+    // initialization step on first call.
+    filtered_orderbook
+        .get_auction_data(ethcontract::U256::zero())
+        .expect("orderbook initialization failed");
```

Comment:
@fleupold What do you think of this? I don't see what advantage we get from doing this in the scheduler. It doesn't feel like there is a right place there because we have multiple schedulers.

Reply:
👍 Yes, that works. It's pretty explicit that it can be a long running operation. I like that `::new` no longer returns a result.

The only slight drawback is that we can't start any of the other background operations before (like price fetching), but since they are fast it shouldn't be an issue.

Reply:
I think the `PriceOracle` is already started at this point and running in a background thread.